pkg/cloudprovider/aws/instance.go

```diff
@@ -104,31 +104,28 @@ func (p *InstanceProvider) Create(ctx context.Context,
 	return createFleetOutput.Instances[0].InstanceIds[0], nil
 }
 
-func (p *InstanceProvider) Terminate(ctx context.Context, nodes []*v1.Node) error {
-	if len(nodes) == 0 {
+func (p *InstanceProvider) Terminate(ctx context.Context, node *v1.Node) error {
+	if node == nil {
 		return nil
 	}
-	ids := p.getInstanceIDs(nodes)
-
-	_, err := p.ec2api.TerminateInstancesWithContext(ctx, &ec2.TerminateInstancesInput{
-		InstanceIds: ids,
+	id, err := p.getInstanceID(node)
+	if err != nil {
+		return fmt.Errorf("getting instance ID for node %s, %w", node.Name, err)
+	}
+	_, err = p.ec2api.TerminateInstancesWithContext(ctx, &ec2.TerminateInstancesInput{
+		InstanceIds: []*string{id},
 	})
 	if err != nil {
```

Comment:
If the instance is not found (cleaned up after it's been terminated), then this will return an error. 

If the instance is terminated, but still present in the console, it will not return an error, but return that it has been "terminated". The same goes for "shutting-down".